### PR TITLE
New filter plugin for syslog security events

### PR DESCRIPTION
**Pull request by mgladi**
A new filter for the OMS Security and Audit solution to collect 3rd party security events from syslog.

### CODE REVIEW

**Comment by robbiezhang** [comment]
@mgladi please add unit tests for the plugin

**Comment by mgladi** [comment]
Changes updated with unit tests for the business logic of the filter

File: source/code/plugins/security_lib.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module Security
```
Comment (robbiezhang) [inline source/code/plugins/security_lib.rb]: module OMS class Security might be more proper.

File: source/code/plugins/filter_syslog_security.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      record.delete 'host'
+
+      host_ip = @ip_cache.get_ip(hostname)
+      if host_ip.nil?
+        OMS::Log.warn_once("Failed to get the IP for #{hostname}.")
+      else
+        record['HostIP'] = host_ip
+      end
+
+      tags = tag.split('.')
```
Comment (robbiezhang) [inline source/code/plugins/filter_syslog_security.rb]: can you add comments about the expected tags?

File: source/code/plugins/security_lib.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    end
+
+    def self.get_data_type(ident)
+      return nil if ident.nil?
+      
+      @@LOG_TYPE_MAPPING.each do |key, data_type|
+        return data_type if ident.start_with?(key)
+      end
+
+      OMS::Log.warn_once("Failed to find data type for record with tag '#{ident}'")
```
Comment (robbiezhang) [inline source/code/plugins/security_lib.rb]: with "ident"?

**Comment by robbiezhang** [comment]
Signoff. You can merge to master once all the minor comments are addressed.

**Pull request merged**